streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -109,16 +111,43 @@ private void buildAggregate(final ProcessorSupplier<K, Change<V>> aggregateSuppl
         final String sinkName = builder.newProcessorName(KStreamImpl.SINK_NAME);
         final String sourceName = builder.newProcessorName(KStreamImpl.SOURCE_NAME);
         final String funcName = builder.newProcessorName(functionName);
+        final String topic = materialized.storeName() + KStreamImpl.REPARTITION_TOPIC_SUFFIX;
+
+        StatefulRepartitionNode.StatefulRepartitionNodeBuilder<K, V, T> statefulRepartitionNodeBuilder = StatefulRepartitionNode.statefulRepartitionNodeBuilder();
 
         buildAggregate(aggregateSupplier,
-                       materialized.storeName() + KStreamImpl.REPARTITION_TOPIC_SUFFIX,
+                       topic,
                        funcName,
-                       sourceName, sinkName);
+                       sourceName,
+                       sinkName
+        );
+
         builder.internalTopologyBuilder.addStateStore(new KeyValueStoreMaterializer<>(materialized)
-                                                              .materialize(), funcName);
+                                                          .materialize(), funcName);
+
+        ProcessorParameters processorParameters = new ProcessorParameters<>(aggregateSupplier, funcName);
+
+        statefulRepartitionNodeBuilder.withRepartitionTopic(topic)
+            .withSinkName(sinkName)
+            .withSourceName(sourceName)
+            .withProcessorParameters(processorParameters)
+            .withKeySerde(keySerde)
+            .withValueSerde(valSerde);
+
+        statefulRepartitionNodeBuilder.withMaterialized(materialized)
```

Comment:
This may take some more thinking, we can pass in a `MaterializedInternal`, `StoreBuilder` or a `String[]` of store names in different places.  While don't disagree about the need for using the`TableProcessorNode` I think we may need to do some general refactoring to consolidate how we represent stateful operations.